Hunting Queries/SecurityEvent/RareProcessesByServiceAccounts.txt

```diff
@@ -0,0 +1,59 @@
+//
+// Id: af02987c-949d-47d5-b0ae-64d8e1b674e2
+//
+// DataSource: #SecurityEvent
+//
+// Tactics: #Persistence, #Discovery, #LateralMovement , #Initial Access, #Execution
+let List1 = datatable(AccountName:string)["MSSQLSERVER", "ReportServer", "MSDTSServer100", "IUSR"];         
+// Provide a list of service account/ built-in accounts in an environment.
+let List2 = SecurityEvent                                                                                   
+// Self generating a list of Service account using event Id :4624
+| where EventID == "4624"
+| where LogonType == "5"                                                                                          
+| extend AccountName = Account 
+| distinct AccountName;
+let Accounts = List1 | union (List2 | distinct AccountName);
+let ProcessCreationEvents=() {
+    let processEvents=SecurityEvent
+    | where EventID==4688
+    // filter out common randomly named files related to MSI installers and browsers
+    | where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{3}\.tmp")
+    | where not(NewProcessName matches regex @"\\TRA[0-9A-Fa-f]{4}\.tmp")
+    | where not(NewProcessName matches regex @"Installer\\MSI[0-9A-Fa-f]{3}\.tmp")
+    | where not(NewProcessName matches regex @"Installer\\MSI[0-9A-Fa-f]{4}\.tmp")
+    | project TimeGenerated, 
+      ComputerName=Computer,
+      AccountName=SubjectUserName, 
+      AccountDomain=SubjectDomainName,
+      FileName=tostring(split(NewProcessName, '\\')[-1]),
+      ProcessCommandLine = CommandLine, 
+      InitiatingProcessFileName=ParentProcessName,
+      InitiatingProcessCommandLine="",
+      InitiatingProcessParentFileName="";
+    processEvents;
+    };
+    let normalizedProcesses = ProcessCreationEvents 
+       // normalize guids
+       | project TimeGenerated, AccountName, ComputerName, ProcessCommandLine, FileName = replace("[0-9A-Fa-f]{8}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{12}", "<guid>", FileName)
+       // normalize digits away
+       | project TimeGenerated, AccountName,ComputerName,   ProcessCommandLine, FileName=replace(@'\d', 'n', FileName); 
+let freqs = normalizedProcesses
+    | summarize frequency = count() by FileName
+    | join kind= leftouter (
+       normalizedProcesses
+       | summarize Since=min(TimeGenerated), LastSeen=max(TimeGenerated)  by FileName, AccountName , ComputerName,  ProcessCommandLine 
+    ) on FileName;
+   let Finalfreqs = freqs 
+    | where frequency <= toscalar( freqs | serialize | project frequency | summarize percentiles(frequency, 25))
+    | order by frequency asc
+    | project FileName, frequency, Since, LastSeen , AccountName , ComputerName,   ProcessCommandLine
+    // restrict results to unusual processes seen in last day 
+    | where LastSeen >= ago(1d);
+Finalfreqs
+    | join kind= inner (
```

Comment:
Since we know the Accounts table should be smaller, it should be on the left, you may want to flip this join operation which will make it more efficient in larger data sets - https://kusto.azurewebsites.net/docs/query/joinoperator.html